nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/metrics/jmx/JmxMetricsResultConverter.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.web.api.metrics.jmx;
+
+import javax.management.openmbean.CompositeData;
+import javax.management.openmbean.TabularData;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class JmxMetricsResultConverter {
+    private static final String COMPOSITE_DATA_KEY = "CompositeData%s";
+
+    public Object convert(final Object attributeValue) {
+            if (attributeValue instanceof CompositeData[]) {
+                final Map<String, Object> values = new HashMap<>();
+                for (int i = 0; i < ((CompositeData[]) attributeValue).length; i++) {
+                    final Map<String, Object> subValues = new HashMap<>();
+                    convertCompositeData(((CompositeData[]) attributeValue)[i], subValues);
+                    values.put(String.format(COMPOSITE_DATA_KEY, i), subValues);
+                }
+                return values;
+            } else if (attributeValue instanceof CompositeData) {
+                final Map<String, Object> values = new HashMap<>();
+                convertCompositeData(((CompositeData) attributeValue), values);
+                return values;
+            } else if (attributeValue instanceof TabularData) {
```

Comment:
What should happen in the value of other attribute value types? The current implementation does not have an `else` condition.